src/components/CatalogOverlay/CatalogOverlayPlotSettingsPanelComponent/CatalogOverlayPlotSettingsPanelComponent.tsx

```diff
@@ -239,74 +241,99 @@ export class CatalogOverlayPlotSettingsPanelComponent extends React.Component<Wi
+                <FormGroup inline={true} disabled={disabledOverlayPanel} label="Reference">
+                    <ButtonGroup>
+                        <AnchorButton onClick={() => widgetStore.setCatalogSizeRef(CatalogSizeRef.CANVAS)} text={CatalogSizeRef.CANVAS} active={widgetStore.catalogSizeRef === CatalogSizeRef.CANVAS} disabled={disabledOverlayPanel} />
+                        <AnchorButton onClick={() => widgetStore.setCatalogSizeRef(CatalogSizeRef.IMAGE)} text={CatalogSizeRef.IMAGE} active={widgetStore.catalogSizeRef === CatalogSizeRef.IMAGE} disabled={disabledOverlayPanel} />
+                        <AnchorButton onClick={() => widgetStore.setCatalogSizeRef(CatalogSizeRef.ANGULAR)} text={CatalogSizeRef.ANGULAR} active={widgetStore.catalogSizeRef === CatalogSizeRef.ANGULAR} disabled={disabledOverlayPanel} />
+                    </ButtonGroup>
+                </FormGroup>
+                <FormGroup inline={true} label="Size" labelInfo={widgetStore.catalogSizeRef === CatalogSizeRef.ANGULAR ? "" : widgetStore.isImagePixelSize ? "(image px)" : "(canvas px)"} disabled={disabledOverlayPanel}>
+                    <Tooltip disabled={disabledOverlayPanel || !widgetStore.disableSizeMap} content={`${widgetStore.minOverlaySize} ~ ${widgetStore.maxOverlaySize}`}>
                         <SafeNumericInput
                             placeholder="Size"
                             disabled={disabledOverlayPanel || !widgetStore.disableSizeMap}
-                            min={CatalogWidgetStore.MinOverlaySize}
-                            max={CatalogWidgetStore.MaxOverlaySize}
+                            min={widgetStore.minOverlaySize}
+                            max={widgetStore.maxOverlaySize}
                             clampValueOnBlur={true}
-                            value={widgetStore.catalogSize}
-                            stepSize={0.5}
+                            value={widgetStore.catalogSize / widgetStore.pixelSizeFactor}
+                            stepSize={0.5 / widgetStore.pixelSizeFactor}
                             onValueChange={(value: number) => widgetStore.setCatalogSize(value)}
                             data-testid="catalog-settings-size-input"
                         />
                     </Tooltip>
+                    <Collapse className="select-angular-unit" isOpen={widgetStore.catalogSizeRef === CatalogSizeRef.ANGULAR}>
+                        <FormGroup inline={true}>
+                            <Select
+                                items={Object.values(AngularSizeUnit)}
+                                activeItem={null}
+                                onItemSelect={units => widgetStore.setFactorToArcsec(units)}
+                                itemRenderer={this.renderUnitPopOver}
+                                disabled={widgetStore.catalogSizeRef !== CatalogSizeRef.ANGULAR}
```

Comment:
When `widgetStore.catalogSizeRef !== CatalogSizeRef.ANGULAR`, it is hidden. I suggest we disable it when the size numeric input is disabled.